### DISCUSSION TITLE
Feedback on Handling of Transition to Matrix 1.1

### DISCUSSION THREAD

**Discussion by synapse-ff40bc8e7455d8130e0ff48b** [element-hq/element-meta#2028] [Product Feedback]
### Your use case

Greetings,

First I want to share my appreciation for the work that you all do in maintaining this client and the contributions to the Matrix ecosystem at large.

The release of `v1.11.40` included PR [#9819](https://github.com/matrix-org/matrix-react-sdk/pull/9819). In short, I feel there's not enough disapproval or voiced frustration in how this change was handled.

Best I can tell, the potential impact of the PR was under-communicated, or I fear worse: not communicated.

With this latest release, those running older and/or legacy versions of synapse are now forced to:

- Use an alternate client,
- Rollback Element to a previous version, or
- Update to a more contemporary version of synapse.

While I think it's fair that operators should not assume that old versions will work in perpetuity, there are situations where operators can not easily update their legacy deployments without going through their own change management processes (e.g. highly regulated environments).

Changes which have the opportunity to impact users should be broadly announced ahead of their release, allowing operators sufficient runway to respond and take action.

This release has likely impacted the business and communication continuity of countless users and made administrators scramble.

**Comment by t3chguy** [comment]
Thanks for the feedback. It isn't specific to Element Desktop, applies to Element Web and all consumers of the [Matrix JS SDK](https://github.com/matrix-org/matrix-js-sdk).
The policy of that project is to guarantee support for 4 spec versions (1 calendar year) - https://github.com/matrix-org/matrix-js-sdk#minimum-matrix-server-version-v11

> Best I can tell, the potential impact of the PR was under-communicated, or I fear worse: not communicated.

![image](https://github.com/vector-im/element-desktop/assets/2403652/f1e75a92-a9dc-4b70-b568-f16b5c148388)

I agree it was under-communicated, definitely something that would have been good to have an additional release note for. cc @andybalaam 

the note for the js-sdk was significantly more prominent:

![image](https://github.com/vector-im/element-desktop/assets/2403652/2b807c13-b263-43c3-a304-f834f96150b7)

This was a change in the making since [this issue](https://github.com/vector-im/element-web/issues/16876).

I suggest https://github.com/vector-im/element-meta/discussions for bringing attention to suggestions around processes and cross-functional issues.

CC @Johennes

**Comment by Johennes** [comment]
Thanks for the initial response, @t3chguy. I've moved and converted this to an element-meta discussion because I'm interested in following up.

@synapse-ff40bc8e7455d8130e0ff48b, in your opinion, what would have been an appropriate form of announcement? Would replicating the JS SDK 🚨 notice into the Element Desktop release notes have been sufficient?

**Comment by clokep** [comment]
https://github.com/matrix-org/synapse/issues/16209 has some additional feedback from a server admin. In particular that the current message is not very actionable.
